babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1583,3 +1583,49 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
             SqlLiteral.createBoolean(withTies, pos));
     }
 }
+
+SqlNode SqlHexCharStringLiteral() :
+{
+    final NlsString p;
+    final String formatString;
+    final String charSet;
+    CharacterSet charSetEnum = null;
+    final HexCharLiteralFormat format;
+}
+{
+    (
+        <PREFIXED_HEX_STRING_LITERAL>
+        {
+            charSet = SqlParserUtil
+                      .trim(SqlParserUtil.getCharacterSet(token.image), " ")
+                      .toUpperCase();
+            if (charSet.equals("LATIN")) {
```

Comment:
nit: Maybe a switch statement would be more appropriate here to avoid having so many "else if"s?

Reply:
Thanks for pointing out, changed at https://github.com/googleinterns/calcite/pull/155/commits/0e2e65e6fa9d39fa4c0b1e62109b8a8ab8f2e15c